packages/strapi-plugin-content-manager/admin/src/components/VariableDraggableAttr/index.js

```diff
@@ -0,0 +1,335 @@
+
+    // Determine mouse position
+    const clientOffset = monitor.getClientOffset();
+
+    // Get pixels to the top
+    const hoverClientY = clientOffset.y - hoverBoundingRect.top;
+
+    // Only perform the move when the mouse has crossed half of the items height
+    // When dragging downwards, only move when the cursor is below 50%
+    // When dragging upwards, only move when the cursor is above 50%
+
+    // Dragging downwards
+    if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
+      return;
+    }
+
+    // Dragging upwards
+    if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
+      return;
+    }
+
+    // Time to actually perform the action
+    props.moveAttr(dragIndex, hoverIndex, props.keys);
+
+    // Note: we're mutating the monitor item here!
+    // Generally it's better to avoid mutations,
+    // but it's good here for the sake of performance
+    // to avoid expensive index searches.
+    monitor.getItem().index = hoverIndex;
+  },
+};
+
+class VariableDraggableAttr extends React.Component {
+  constructor(props) {
+    super(props);
+    const { data, layout, name } = this.props;
+    const appearance = get(layout, [name, 'appearance'], '');
+    const type = appearance !== '' ? appearance : data.type;    
+    let classNames = getBootstrapClass(type);
+    let style = {};
+
+    if (!type) {
+      style = { display: 'none' };
+      classNames = {
+        bootstrap: name.split('__')[1],
+        wrapper: cn(styles.attrWrapper),
+        withLongerHeight: false,
+      };
+    }
+    this.state = {
+      classNames,
+      dragStart: false,
+      isOver: false,
+      style,
+    };
+  }
+
+  componentDidMount() {
+    // Use empty image as a drag preview so browsers don't draw it
+    // and we can draw whatever we want on the custom drag layer instead.
+    this.props.connectDragPreview(getEmptyImage(), {
+    // IE fallback: specify that we'd rather screenshot the node
+    // when it already knows it's being dragged so we can hide it with CSS.
+    // Removginv the fallabck makes it handle variable height elements
+      // captureDraggingState: true,
+    });
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isDragging !== this.props.isDragging) {
+      this.handleDragEffect();
+    }
+
+    if (prevProps.isDragging !== this.props.isDragging && this.props.isDragging) {
+      this.handleClickEdit();
+    }
+  }
+
+  handleClickEdit = () => {
+    this.props.onClickEdit(this.props.index);
+  }
+
+  handleDragEffect = () => this.setState(prevState => ({ dragStart: !prevState.dragStart }));
+
+  handleMouseEnter= () => {
+    this.setState({ isOver: true });
+  }
+
+  handleMouseLeave = () => this.setState({ isOver: false });
+
+  handleRemove = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    const { index, keys, onRemove } = this.props;
+    onRemove(index, keys);
+  }
+
+  renderContent = () => {
```

Comment:
Please add comments and rename some variables with more explicit names like `itemLineEls` because it's very complicated this method :/

Reply:
Done